lib/src/layer/polygon_layer/polygon_layer.dart

```diff
@@ -64,50 +66,57 @@ class PolygonLayer extends StatefulWidget {
 }
 
 class _PolygonLayerState extends State<PolygonLayer> {
-  final _cachedSimplifiedPolygons = <int, List<Polygon>>{};
+  List<_ProjectedPolygon>? _cachedProjectedPolygons;
+  final _cachedSimplifiedPolygons = <int, List<_ProjectedPolygon>>{};
 
   @override
   void didUpdateWidget(PolygonLayer oldWidget) {
     super.didUpdateWidget(oldWidget);
 
-    // IF old yes & new no, clear
-    // IF old no & new yes, compute
-    // IF old no & new no, nothing
-    // IF old yes & new yes & (different tolerance | different lines), both
-    //    otherwise, nothing
-    if (oldWidget.simplificationTolerance != 0 &&
-        widget.simplificationTolerance != 0 &&
-        (!listEquals(oldWidget.polygons, widget.polygons) ||
-            oldWidget.simplificationTolerance !=
-                widget.simplificationTolerance)) {
-      _cachedSimplifiedPolygons.clear();
-      _computeZoomLevelSimplification(MapCamera.of(context).zoom.floor());
-    } else if (oldWidget.simplificationTolerance != 0 &&
-        widget.simplificationTolerance == 0) {
-      _cachedSimplifiedPolygons.clear();
-    } else if (oldWidget.simplificationTolerance == 0 &&
-        widget.simplificationTolerance != 0) {
-      _computeZoomLevelSimplification(MapCamera.of(context).zoom.floor());
+    if (widget.simplificationTolerance != 0 &&
+        oldWidget.simplificationTolerance == widget.simplificationTolerance &&
+        listEquals(oldWidget.polygons, widget.polygons)) {
+      // Reuse cache.
+      return;
     }
+
+    _cachedSimplifiedPolygons.clear();
+    _cachedProjectedPolygons = null;
```

Comment:
This logic has been broken (changing the tolerance no longer 'refreshes' the layer correctly/at all).

Reply:
changing the tolerance will clear the cache. Am I missing something?

Reply:
Try dragging the slider in the example app. It doesn't change the tolerance after the slider is released. That's why I called the calculation methods here before - although there may be a better fix.

Reply:
I see. Btw I hadn't seen the sress test before - cool stuff.

The logic works correctly, however the units changed and therefore absolute distances. In other words [0; 2] is now too tight of a tollerance. I'd suggest a more reasonable new default tolerance of ~150,000

Reply:
I see. Why has the unit changed, and what is it now? Is there any way we can change it back? We were in-keeping with other libraries units and recommended ranges.

Reply:
we went for X a.k.a longitude from (-180; 180) to whatever the projection spits out (similar for Y a.k.a latitude).

... as I type it out I do acknowledge that at this point a good default would be CRS dependent. So it might be best to derive an effective tollerance instead based on the projection. Should be easy.

Reply:
I'm assuming you say that because it depends on how close two points are once projected now (and projection can 'move' the points now)? Is that a decent understanding?

If so, I'm good with that, it's not like the layers have a `const` constructor that would be ruined. However, we need to find a succinct, relatively non-technical way to explain the unit to the user. It was previously "decimal degrees scaled by floored zoom" I think, which still isn't perfect, but at least you can "feel" what the numbers do as it's a small scale: it would be good to keep this small scale.

Reply:
Yeah I don't feel that unit either, but you could get a feel for the scale :) 0.5 was like x2 simplification, 1 was x4, 2 was x8 IMO.

Yes, I think pixel distances would be ideal. That would 'remove' the need for scaling the tolerance value based on zoom level, as that would happen inherently. I suppose thats also now easier that the polylines/gons are in pixel space already now.
The best approach would also be to take into account the thickness of the line (a really thin line will 'show' more simplification than a wide one), but that might be too complex to implement and difficult to calibrate, so ignore that.

Reply:
> I suppose thats also now easier that the polylines/gons are in pixel space already now.

Yes. We're almost in pixel space. If you'd replace the "project" with "project+transform" in the effective tolerance calculation that's what you'd get.

> The best approach would also be to take into account the thickness of the line (a really thin line will 'show' more simplification than a wide one), but that might be too complex to implement and difficult to calibrate, so ignore that.

Should be fairly easy to do. But maybe in a follow-up? For now the tolerances should behave fairly similar to what they were before.

There's other things one could do in pixel space, e.g. if the polygon on screen is only 1x1 pixel in size, why even simplify just cull the thing, etc.

Reply:
> if the polygon on screen is only 1x1 pixel in size

That should be rare enough that it's not an issue :D

> Should be fairly easy to do. But maybe in a follow-up? For now the tolerances should behave fairly similar to what they were before.

Yeah, don't worry about this extension for now. But besides that, using pixel tolerances for simplification should be done here, as that seems like a very good idea and you say we're almost there?

Reply:
> That should be rare enough that it's not an issue :D

In many ways simplification already helps with that case. Whether or not it's rare :shrug:

![Screenshot from 2024-01-17 13-32-17](https://github.com/fleaflet/flutter_map/assets/111986/20632547-d97f-4034-afac-8f7f3eeb00f2)

> Yeah, don't worry about this extension for now. But besides that, using pixel tolerances for simplification should be done here, as that seems like a very good idea and you say we're almost there?

Done.

Reply:
Great, thanks. I'll have a play around in a few hours and choose a new sensible default (assuming it needs changing now that the units have changed).